src/main/java/org/prebid/server/bidder/colossus/ColossusBidder.java

```diff
@@ -1,20 +1,143 @@
+        }
+    }
+
+    private static Imp modifyImp(Imp imp, String tagId) {
+        return imp.toBuilder().tagid(tagId).build();
+    }
+
+    private HttpRequest<BidRequest> makeRequest(BidRequest bidRequest, Imp imp) {
+        final BidRequest outgoingRequest = bidRequest.toBuilder().imp(Collections.singletonList(imp)).build();
+
+        return HttpRequest.<BidRequest>builder()
+                .method(HttpMethod.POST)
+                .headers(HttpUtil.headers())
+                .uri(endpointUrl)
+                .body(mapper.encodeToBytes(outgoingRequest))
+                .payload(outgoingRequest)
                 .build();
     }
+
+    @Override
+    public Result<List<BidderBid>> makeBids(HttpCall<BidRequest> httpCall, BidRequest bidRequest) {
+        final BidResponse bidResponse;
+        try {
+            bidResponse = mapper.decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
+        } catch (DecodeException e) {
+            return Result.withError(BidderError.badServerResponse(e.getMessage()));
+        }
+
+        final List<BidderError> errors = new ArrayList<>();
+        final List<BidderBid> bids = extractBids(httpCall.getRequest().getPayload(), bidResponse, errors);
+
+        return Result.of(bids, errors);
+    }
+
+    private static List<BidderBid> extractBids(BidRequest bidRequest,
+                                               BidResponse bidResponse,
+                                               List<BidderError> errors) {
+
+        if (bidResponse == null || CollectionUtils.isEmpty(bidResponse.getSeatbid())) {
+            return Collections.emptyList();
+        }
+
+        return bidResponse.getSeatbid().stream()
+                .filter(Objects::nonNull)
+                .map(SeatBid::getBid)
+                .filter(Objects::nonNull)
+                .flatMap(Collection::stream)
+                .filter(Objects::nonNull)
+                .map(bid -> makeBidderBid(bid, bidRequest.getImp(), bidResponse.getCur(), errors))
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+    }
+
+    private static BidderBid makeBidderBid(Bid bid, List<Imp> imps, String currency, List<BidderError> errors) {
+        try {
+            return BidderBid.of(bid, resolveBidType(bid.getImpid(), imps), currency);
+        } catch (PreBidException e) {
+            errors.add(BidderError.badServerResponse(e.getMessage()));
+            return null;
+        }
+    }
+
+    private static BidType resolveBidType(String impId, List<Imp> imps) {
+        for (Imp imp : imps) {
+            if (Objects.equals(impId, imp.getId())) {
+                if (imp.getBanner() == null && imp.getVideo() != null) {
+                    return BidType.video;
+                }
+                return BidType.banner;
+            }
+        }
+
+        throw new PreBidException(String.format("Failed to find impression for ID: %s", impId));
```

Comment:
Seems like this logic does not exactly match logic present in OpenrtbBidder

Reply:
Pls, consider go code